File: src/pocketcloud/network/packet/impl/normal/PlayerSyncPacket.php
<?php

namespace pocketcloud\network\packet\impl\normal;

use pocketcloud\network\client\ServerClient;
use pocketcloud\network\packet\CloudPacket;
use pocketcloud\network\packet\utils\PacketData;
use pocketcloud\player\CloudPlayer;

class PlayerSyncPacket extends CloudPacket {

    public function __construct(
        private ?CloudPlayer $player = null,
        private bool $removal = false
    ) {}

    public function encodePayload(PacketData $packetData) {
        $packetData->writePlayer($this->player);
        $packetData->write($this->removal);
    }

    public function decodePayload(PacketData $packetData) {
        $this->player = $packetData->readPlayer();
        $this->removal = $packetData->readBool();
    }

    public function getPlayer(): ?CloudPlayer {
        return $this->player;
    }

    public function isRemoval(): bool {
        return $this->removal;
    }

    public function handle(ServerClient $client) {}
}

File: cloud-network/src/main/java/de/pocketcloud/network/packet/impl/PlayerSyncPacket.java
package de.pocketcloud.network.packet.impl;

import de.pocketcloud.api.model.player.ICloudPlayer;
import de.pocketcloud.api.network.packet.ClientboundPacket;
import de.pocketcloud.api.network.packet.data.IPacketData;
import de.pocketcloud.common.mapper.MapperUtils;
import de.pocketcloud.network.packet.CloudPacket;
import lombok.Getter;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@Getter
public final class PlayerSyncPacket extends CloudPacket implements ClientboundPacket {

    private ICloudPlayer player;
    private boolean removal;

    public PlayerSyncPacket(ICloudPlayer player, boolean removal) {
        this.player = player;
        this.removal = removal;
    }

    @Override
    public void encodePayload(IPacketData packetData) {
        packetData.writeAll(MapperUtils.toMap(player), removal);
    }

    @Override
    public void decodePayload(IPacketData packetData) {}

    public static PlayerSyncPacket create(ICloudPlayer player, boolean removal) {
        return new PlayerSyncPacket(player, removal);
    }
}
